test/e2e/certman_operator_tests.go

```diff
@@ -60,6 +97,116 @@ var _ = Describe("Certman Operator", Ordered, func() {
 				return false
 			}
 			return apiserver.Spec.ServingCerts.NamedCertificates[0].ServingCertificate.Name == secretName
-		}, pollingDuration, 30*time.Second).Should(BeTrue(), "Certificate secret should be applied to apiserver object")
+		}, pollingDuration, 30*time.Second).Should(gomega.BeTrue(), "Certificate secret should be applied to apiserver object")
 	})
+	ginkgo.It("Performs AWS secret deletion scenario end-to-end", func(ctx context.Context) {
+		ginkgo.By("ensuring AWS secret exists")
+		gomega.Eventually(func() bool {
+			secret, err := clientset.CoreV1().Secrets(operatorNS).Get(ctx, awsSecretName, metav1.GetOptions{})
+			if err != nil {
+				return false
+			}
+			awsSecretBackup = secret.DeepCopy()
+			return true
+		}, pollingDuration, 30*time.Second).Should(gomega.BeTrue(), "AWS secret does not exist")
+
+		ginkgo.By("deleting AWS secret")
+		err := clientset.CoreV1().Secrets(operatorNS).Delete(ctx, awsSecretName, metav1.DeleteOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred(), "Failed to delete AWS secret")
+
+		ginkgo.By("waiting for operator pod to be stable after AWS secret deletion")
+		gomega.Eventually(func() bool {
+			pods, err := clientset.CoreV1().Pods(operatorNS).List(ctx, metav1.ListOptions{})
+			if err != nil || len(pods.Items) == 0 {
+				return false
+			}
+			for _, pod := range pods.Items {
+				if strings.Contains(pod.Name, "certman-operator") {
+					if pod.Status.Phase != corev1.PodRunning {
+						return false
+					}
+					if len(pod.Status.ContainerStatuses) == 0 || pod.Status.ContainerStatuses[0].RestartCount != 0 {
```

Comment:
what is the check here? if we would like the container to be running without restart. Should we check 1) the length of the `pod.Status.ContainerStatuses != 1`, 2) `pod.Status.ContainerStatuses[0].Ready != true` ?

We can merge this now, but I think you need to review the logic here.